packages/core/src/odata-common/entity-builder.ts

```diff
@@ -82,3 +116,30 @@ export class EntityBuilder<EntityT extends Entity, JsonT> {
 function toClassName(entityName: string) {
   return entityName.substr(entityName.indexOf('_') + 1);
 }
+
+function buildNavigationPropertyFromJson<
+  EntityT extends Entity,
+  LinkedEntityT extends Entity
+>(
+  key: string,
+  value: FromJsonType<unknown>,
+  entityConstructor: Constructable<EntityT>
+): LinkedEntityT | LinkedEntityT[] {
+  const field = entityConstructor[upperCaseSnakeCase(key)];
+  const linkedEntityConstructor: Constructable<LinkedEntityT> =
+    field._linkedEntity;
+  return Array.isArray(value)
+    ? value.map(item =>
+        buildSingleEntityFromJson(item, linkedEntityConstructor)
+      )
+    : buildSingleEntityFromJson(value, linkedEntityConstructor);
+}
+
+function buildSingleEntityFromJson<LinkedEntityT extends Entity>(
+  json: FromJsonType<unknown>,
+  linkedEntityConstructor: Constructable<LinkedEntityT>
+): LinkedEntityT {
+  return json instanceof linkedEntityConstructor
```

Comment:
Is this correct? From a quick look I would guess that if it is a `linkedEntityConstructor ` you can not return it directly but need to call a builder?

Reply:
It is a little awkward, but we are looking at `instanceof` and the constructor is basically the class.